controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -393,6 +418,9 @@ func (pp *portPublisher) endpointsToAddresses(endpoints *corev1.Endpoints) PodSe
 	for _, subset := range endpoints.Subsets {
 		resolvedPort := pp.resolveTargetPort(subset)
 		for _, endpoint := range subset.Addresses {
+			if pp.hostname != "" && pp.hostname != endpoint.Hostname {
```

Comment:
why the `pp.hostname != ""` check? if `pp.hostname` == "" and `endpoint.Hostname` == "foo" this check will fail, is that the intent?

Reply:
If the portPublisher's hostname is unset (empty string) that means that no hostname was specified.  In other words, this is a regular service lookup and not a stable pod id lookup.  Therefore, we do not want to skip any of the endpoints.

Reply:
So does that mean `pp.hostname == "" && endpoint.Hostname == "foo"` will never happen? And if it does, then there is a bug elsewhere? Or can we just do `if pp.hostname != endpoint.Hostname` here?

Reply:
`pp.hostname` corresponds to the subscriber's query whereas `endpoint.Hostname` corresponds to the hostname of the pod.  So, for example, if we did a destination lookup for `foo.default.svc.cluster.local` where `foo` is a service which selects a stateful set, then `pp.hostname` would be empty (because we are querying the entire service, not a particular stable id) but each pod in the service would have a non-empty `endpoint.Hostname`.

In this case, we want to select all the pods, and therefore do not want to hit the `continue`.